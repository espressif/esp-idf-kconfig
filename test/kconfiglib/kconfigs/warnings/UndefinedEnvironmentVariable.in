mainmenu "test using simple macros"
# Will not work with upstream kconfiglib as it supports only "${NAME}" expansion

# This macro should be ignored
MAX_NUMBER_OF_MOTORS = 8


config SECOND_MOTOR_NUMBER
    int "Second number of motors"
    default ${UNDEFINED_MAX_NUMBER_OF_MOTORS} # should expand to 4 (envvar set in test_kconfiglib.py)
    help
        Number of motors connected to the starship

config SECOND_MOTOR_NUMBER_AS_STRING
    string "Second motor number as string"
    default "${UNDEFINED_MAX_NUMBER_OF_MOTORS}" # should expand to "4" (envvar set in test_kconfiglib.py)
