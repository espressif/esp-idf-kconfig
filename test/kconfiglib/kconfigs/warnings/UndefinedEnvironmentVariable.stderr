the int symbol SECOND_MOTOR_NUMBER
has a non-int default ${UNDEFINED_MAX_NUMBER_OF_MOTORS} (undefined)
